### ISSUE TITLE
Upgrade rdoc dependency to 5.X

### ISSUE THREAD

**Issue by Azolo** [vertiginous/rdoc_chm#7]
This requires looking at and editing the `Darkfish` template overrides to match the ones that `rdoc 4.X` uses or possible look at ways to write something similar that doesn't depend on `Darkfish`.

**Comment by Azolo** [comment]
@zzak, @windwiny This is a ticket resulting from rdoc/rdoc#390

**Comment by zzak** [comment]
Ok! I will look into this

**Comment by vais** [comment]
@zzak bump, did you happen to make any progress on this?

@Azolo if this turns out to be an intractable issue, my 2c would be to drop this chm help feature from https://github.com/oneclick/rubyinstaller altogether. It may have made sense at the time, but, as a 10-year user of Ruby on Windows, I did not even know chm help was ever available. Not to mention it's a sub-par experience compared to just Googling, which is what most newcomers to Ruby will be doing anyway. IMHO CHM is at best a vestigial feature for the RubyInstaller project. If it's holding up releases for months (!), maybe it's time to just let it go?

**Comment by vais** [comment]
@Azolo to add to the above, I would like to start running https://github.com/ruby/spec against 2.3.1 on Windows, but for that I need to get AppVeyor to add 2.3.1 to this list https://www.appveyor.com/docs/installed-software#ruby. It's a tall stack of yaks and CHM is standing in the way of progress dammit :stuck_out_tongue_winking_eye:

**Comment by zzak** [comment]
@vais Yeah, sorry I haven't had time to look into this yet but your suggestion might be worth considering!

**Comment by Azolo** [comment]
@zzak Continuation from oneclick/rubyinstaller#348

I know we talked about this before, and thanks again for your offer. If you don't get around to looking at it don't sweat it too much. It's not on you to make this happen. :smile:

Here is some more context, for you or hopefully some other adventurous soul.

I think that it's really not as much about `rdoc` as the fact `rdoc_chm` depends so heavily on the `Darkfish` generator. A big problem is that I don't understand enough about what `Darkfish` is doing or what need to be done.

What would really help out is if you could tell me, in your opinion, if it would be easier just to forsake trying to inherit from `Darkfish` and replicate the relevant parts or try to figure out what need to be changed to keep using Darkfish to power the guts of the CHM generator.

**Comment by Azolo** [comment]
@vais I get you, and before I started maintaining RubyInstaller I thought the same thing. But I do know that people use it and once I tried it I released that the experience is actually pretty good.

When I don't have an internet connection or when I know I'm going to be jumping around the documentation quite a bit, I actually find the CHM to be quite useful. But that's really not often these days.

Plus, it actually generates PRs on RubyInstaller to make it better more often than you would think.
